hail/src/main/scala/is/hail/expr/ir/IsScalarConstant.scala

```diff
@@ -5,18 +5,16 @@ import is.hail.expr.types._
 object IsScalarType {
   def apply(t: Type): Boolean = {
     t match {
-      case TInt32(_) | TInt64(_) | TFloat32(_) | TFloat64(_) | TBoolean(_) => true
+      case TInt32(_) | TInt64(_) | TFloat32(_) | TFloat64(_) | TBoolean(_) | TString(_) => true
       case _ => false
     }
   }
 }
 
-
 object IsScalarConstant {
   def apply(ir: IR): Boolean = {
     ir match {
-      case I32(_) | I64(_) | F32(_) | F64(_) | True() | False() => true
-      case NA(t) if IsScalarType(t) => true
+      case I32(_) | I64(_) | F32(_) | F64(_) | True() | False() | NA(_) | Literal(_, _) | Die(_, _) => true
```

Comment:
the ones you've added are constant, but they're not all scalars,  i don't think. Are you changing this to just mean "IsConstant"?

Reply:
yes, good point